auth0/resource_auth0_user.go

```diff
@@ -0,0 +1,163 @@
+package auth0
+
+import (
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/yieldr/go-auth0/management"
+)
+
+func newUser() *schema.Resource {
+	return &schema.Resource{
+		Create: createUser,
+		Read:   readUser,
+		Update: updateUser,
+		Delete: deleteUser,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"user_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+				StateFunc: func(val interface{}) string {
+					return "auth0|" + val.(string)
+				},
+			},
+			"conn": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"email": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"username": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"password": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"phone_number": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"user_metadata": {
+				Type:     schema.TypeMap,
+				Optional: true,
+			},
+			"app_metadata": {
+				Type:     schema.TypeMap,
+				Optional: true,
+			},
+			"email_verified": {
+				Type:     schema.TypeBool,
+				Optional: true,
+			},
+			"verify_email": {
+				Type:     schema.TypeBool,
+				Optional: true,
+			},
+			"phone_verified": {
+				Type:     schema.TypeBool,
+				Optional: true,
+			},
+		},
+	}
+}
+
+func readUser(d *schema.ResourceData, m interface{}) error {
+	api := m.(*management.Management)
+	u, err := api.User.Read(d.Id())
+	if err != nil {
+		return err
+	}
+
+	for k, v := range map[string]interface{}{
+		"user_id":        u.ID,
+		"conn":           u.Connection,
+		"username":       u.Username,
+		"phone_number":   u.PhoneNumber,
+		"user_metadata":  u.UserMetadata,
+		"email_verified": u.EmailVerified,
+		"phone_verified": u.PhoneVerified,
+		"verify_email":   u.VerifyEmail,
+		"app_metadata":   u.AppMetadata,
+		"email":          u.Email,
+		"password":       u.Password,
+	} {
+		if (k == "password" || k == "conn") && v.(*string) == nil {
+			continue
+		}
+
+		if err := d.Set(k, v); err != nil {
+			return err
+		}
+	}
```

Comment:
Which one do you mean ?

Reply:
Any one of [`updateXXX`](https://github.com/yieldr/terraform-provider-auth0/blob/d0a2ce6d27a0fecb734f588ca945636288692981/auth0/resource_auth0_client.go#L274) and [`buildXXX`](https://github.com/yieldr/terraform-provider-auth0/blob/d0a2ce6d27a0fecb734f588ca945636288692981/auth0/resource_auth0_client.go#L289) for example.

Reply:
I'm sorry but I still don't get the point, you mean why update, build and read don't follow conventions you defined ? If so which convention exactly ?

Reply:
Yes exactly, the convention I just linked. Comparing the two seems very different. All the resources follow this convention and I'd prefer to keep that.

Reply:
It's different because you don't check the error coming from Set function. I can remove this but it's not a good thing not checking errors coming from a function, it could lead to unexpected troubles hard to debug.